declarative_tree.py

```diff
@@ -0,0 +1,220 @@
+
+from __future__ import annotations
+from sympy import Symbol, Not, Expr, Or, And
+from sympy.logic.boolalg import simplify_logic, to_dnf, BooleanAtom
+import itertools as it
+from typing import Dict, List, Set, Optional, Union, Callable, Tuple
+from dataclasses import dataclass
+
+class Contains(Symbol):
+    def __init__(self, *args):
+        super().__init__()
+        self.prob=0.1
+
+    def eval(self, string):
```

Comment:
as above. we need a `.lower()` here or we conform to a standard lowercase.

Reply:
I like how your comments stay here so you just look insane